cmd/crc/cmd/daemon_nonwin.go

```diff
@@ -13,7 +13,7 @@ import (
 func runDaemon() {
 	// Remove if an old socket is present
 	os.Remove(constants.DaemonSocketPath)
-	crcAPIServer, err := api.CreateAPIServer(constants.DaemonSocketPath, newViperConfig)
+	crcAPIServer, err := api.CreateAPIServer(constants.DaemonSocketPath, newViperConfig, api.NewHandler())
```

Comment:
For `CreateAPIServer` you will always be passing `api.NewHandler()`, the tests use `createAPIWithListener` directly, maybe you don't need to add the `handler` argumen to `CreateAPIServer` and only add it to `CreateAPIWithListener`?

Reply:
Done!